22.2.3+ (#1)
friedcucumber/maui-grid-kanban-view: README.md

## Implementation Details

> * You can handle the [DataGridView.CustomSort](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataGrid.DataGridView.CustomSort) event to sort groups in the order you want:

Reviewer: In this context, "order" works like "mode". Besides, order I want can be any -- actually one of two, that is, ascending or descending.

```suggestion
* You can handle the [DataGridView.CustomSort](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataGrid.DataGridView.CustomSort) event to sort groups in ascending or descending order:
```

    ```csharp
    private void DataGridView_CustomSort(object sender, DevExpress.Maui.DataGrid.CustomSortEventArgs e) {
        if (e.Column.FieldName == "Stage")
            e.Result = Comparer.Default.Compare(viewModel.StageOrder[(string)e.Value1], viewModel.StageOrder[(string)e.Value2]);
    }
    ```
    File to Look At: [MainPage.xaml.cs](CS/MainPage.xaml.cs)

> * If a [Data Grid](https://docs.devexpress.com/MAUI/403255/data-grid/data-grid) group does not contain any item, the placeholder (based on the [Border](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/border) control) is displayed. To achive this, we handle the [CompleteRowDragDrop](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataGrid.DataGridView.CompleteRowDragDrop) event and switch the `IsPlaceholder` property. The placeholder is displayed when the `IsPlaceholder` property is `true`. .NET MAUI [triggers](https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/triggers) allow you to conditionally display a specific element:

Reviewer: > If a [Data Grid](https://docs.devexpress.com/MAUI/403255/data-grid/data-grid) group **does not contain any item**

This phrase sounds somewhat strange, don't you think? Surely, we can say as follows:

> If a Data Grid group does not contain any items

Yet still, isn't there a simpler and shorter way to say the same? There is:

> If a Data Grid group **is empty** ...

Much simpler, don't you agree?

> **the** placeholder (based on the [Border](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/border) control) **is displayed**.

What do we have here -- two things -- an incorrect article case and avoidable passive voice.

Why "placeholder" should be prepended with an indefinite article? Because we haven't introduced the concept of "placeholder" in this topic earlier on.

Finally, passive voice. To get rid of it, we need to rephrase an entire sentence, taking into account who displays the mentioned placeholder. And that would be nothing else than "Data Grid"... And we have the following sentence in the end:

> If a group is empty, [Data Grid](https://docs.devexpress.com/MAUI/403255/data-grid/data-grid) displays a placeholder based on the [Border](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/border) control.

As you may notice, we also got rid of the bracketed clause and turned it into an addendum.

> **To achive this, we** handle the [CompleteRowDragDrop](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataGrid.DataGridView.CompleteRowDragDrop) event and switch the `IsPlaceholder` property. The placeholder is displayed when the `IsPlaceholder` property is `true`.

First -- achive instead of "achieve" -- a typo.

Second -- to achieve what? To display a placeholder. But we already said that Data Grid displays a placeholder instead of a group when it's empty. We'll remember this fact and adjust the first sentence later on.

Third, "we". I have no objections against the pronoun "we" per se. However, we already addressed our reader as "you" and urged "them" to perform all these actions required to accomplish the goals of this code example. Therefore, I wouldn't mentioned "us" for consistency and stuck to simple imperative instructions - do this, do that, handle X, etc.

> **Switch** the `IsPlaceholder` property

Yes, we probably can say "switch" in this context because it implies interaction with a Boolean property as a toggle switch. However, I would have preferred a more explicit form, such as set the `IsPlaceholder` property to `true`. But this form is longer, true. There is one more alternative -- to say "to enable the `IsPlaceholder` property". This form adds more clarity compared to "switch" but goes by with the same amount of words, which suits us.

> The placeholder is displayed when the `IsPlaceholder` property is `true`.

This sentence actually repeats the idea we've just said in the previous sentence. Therefore, I would have just removed this sentence and be done with it.

> .NET MAUI [triggers](https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/triggers) allow you to conditionally display a specific element:

Unfortunately, with the current wording, I don't feel connection with the rest of this paragraph. Does .NET MAUI allows me to conditionally display a specific element instead of the mentioned placeholder? Instead of an empty group? Once I read this sentence, I ended up with more questions than I had before starting to read this paragraph.

I will rewrite this paragraph according to my reasoning here, however, you'll have to deal with the last sentence here:

```suggestion
* If a group is empty, [Data Grid](https://docs.devexpress.com/MAUI/403255/data-grid/data-grid) can display a placeholder based on the [Border](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/border) control. To display placeholders, handle the [CompleteRowDragDrop](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataGrid.DataGridView.CompleteRowDragDrop) event and enable the `IsPlaceholder` property. .NET MAUI [triggers](https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/triggers) allow you to conditionally display a specific element:
```

Even with these changes, the last sentence in this paragraph does not work, and I lack the knowledge required to make it work. We need to resolve this issue before moving forward.
